### PR TITLE
Onia2mumu

### PR DESCRIPTION
**Pull request by alberto-sanchez**
Code for BPH compactSKIM. We add Onia2MuMu package, low energy photon conversion
selector package and the cff file for skimming.

### CODE REVIEW

**Comment by slava77** [comment]
Please rename compactSKIM to CompactSkim to follow package naming style

**Comment by slava77** [comment]
Sorry for piecemeal review:  
HeavyFlavorAnalysis/CompactSkim has only two files. It's not meaningful to make a new package for this. Please move the files to  HeavyFlavorAnalysis/Skimming

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by davidlange6** [comment]
@alberto-sanchez - thanks for this

a few small items - 
-- put everything in one package (Onia2MuMu seems best)
-- consumes migration - getByToken instead of getByLabel
-- The PhotonConversionProducer and VertexReProducer class names are too generic for this package. Please either change their name (add Onia?) or put them in a namespace (maybe easier)

**Comment by alberto-sanchez** [comment]
@davidlange6  - I have uploaded the required modifications and tested them.

**Comment by davidlange6** [comment]
@cmsbuild, please test

File: HeavyFlavorAnalysis/Onia2MuMu/interface/Onia2MuMuPAT.h [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+#include <CommonTools/UtilAlgos/interface/StringCutObjectSelector.h>
+#include "RecoVertex/VertexTools/interface/InvariantMassFromVertex.h"
+
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+
+template<typename T>
+struct GreaterByVProb {
+  typedef T first_argument_type;
+  typedef T second_argument_type;
```
Comment (davidlange6) [inline HeavyFlavorAnalysis/Onia2MuMu/interface/Onia2MuMuPAT.h]: @alberto-sanchez - I guess that these typedefs are not needed? They seem not to be used.

File: HeavyFlavorAnalysis/Onia2MuMu/interface/Onia2MuMuPAT.h [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    return t1.userFloat("vProb") > t2.userFloat("vProb");
+  }
+};
+
+
+//
+// class decleration
+//
+
+class Onia2MuMuPAT : public edm::EDProducer {
```
Comment (davidlange6) [inline HeavyFlavorAnalysis/Onia2MuMu/interface/Onia2MuMuPAT.h]: @alberto-sanchez 
I missed this before - I think this module can become a edm::stream::EDProducer<>
?

**Comment by davidlange6** [comment]
@slava77, @montjj - do you see any remaining issues with this PR?

**Comment by slava77** [comment]
@davidlange6 
I haven't really looked (not a proper reco code)

File: HeavyFlavorAnalysis/Onia2MuMu/interface/Onia2MuMuPAT.h (hunk 1 above)
Comment (alberto-sanchez) [inline HeavyFlavorAnalysis/Onia2MuMu/interface/Onia2MuMuPAT.h]: Thanks for finding this. I have remove those.

File: HeavyFlavorAnalysis/Onia2MuMu/interface/Onia2MuMuPAT.h (hunk 2 above)
Comment (alberto-sanchez) [inline HeavyFlavorAnalysis/Onia2MuMu/interface/Onia2MuMuPAT.h]: @davidlange6 
Sorry for my ignorance. Could please elaborate a bit?. I was not even aware of edm::stream::EDProducer<>

**Comment by monttj** [comment]
+1
I don't have any specific issue on this package.

**Comment by davidlange6** [comment]
@degano - could you assign this package to analysis tools? THanks!

**Pull request merged**